nvflare/app_opt/job_launcher/process_launcher.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import shlex
+import subprocess
+import sys
+
+from nvflare.apis.fl_constant import FLContextKey, JobConstants
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.workspace import Workspace
+from nvflare.app_opt.job_launcher.job_launcher_spec import JobHandleSpec, JobLauncherSpec
+from nvflare.private.fed.utils.fed_utils import add_custom_dir_to_path
+
+
+class ProcessHandle(JobHandleSpec):
+    def __init__(self, process):
+        super().__init__()
+
+        self.process = process
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def terminate(self):
+        if self.process:
+            try:
+                os.killpg(os.getpgid(self.process.pid), 9)
+                self.logger.debug("kill signal sent")
+            except:
+                pass
+
+            self.process.terminate()
+
+    def poll(self):
+        if self.process:
+            return self.process.poll()
```

Comment:
We need to map to a set of standard return codes. Otherwise how can you manage the return codes from the upper layer?

Reply:
standardized the return code.